modules/nf-core/dysgu/main.nf

```diff
@@ -0,0 +1,91 @@
+//               https://github.com/nf-core/modules/tree/master/modules/nf-core/
+//               You can also ask for help via your pull request or on the #modules channel on the nf-core Slack workspace:
+//               https://nf-co.re/join
+// TODO nf-core: A module file SHOULD only define input and output files as command-line parameters.
+//               All other parameters MUST be provided using the "task.ext" directive, see here:
+//               https://www.nextflow.io/docs/latest/process.html#ext
+//               where "task.ext" is a string.
+//               Any parameters that need to be evaluated in the context of a particular sample
+//               e.g. single-end/paired-end data MUST also be defined and evaluated appropriately.
+// TODO nf-core: Software that can be piped together SHOULD be added to separate module files
+//               unless there is a run-time, storage advantage in implementing in this way
+//               e.g. it's ok to have a single module for bwa to output BAM instead of SAM:
+//                 bwa mem | samtools view -B -T ref.fasta
+// TODO nf-core: Optional inputs are not currently supported by Nextflow. However, using an empty
+//               list (`[]`) instead of a file can be used to work around this issue.
+
+process DYSGU {
+    tag "$meta.id"
+    label 'process_medium'
+
+    // TODO nf-core: List required Conda package(s).
+    //               Software MUST be pinned to channel (i.e. "bioconda"), version (i.e. "1.10").
+    //               For Conda, the build (i.e. "h9402c20_2") must be EXCLUDED to support installation on different operating systems.
+    // TODO nf-core: See section in main README for further information regarding finding and adding container addresses to the section below.
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/dysgu:latest':
+        'biocontainers/dysgu:latest' }"
+
+    input:
+    // TODO nf-core: Where applicable all sample-specific information e.g. "id", "single_end", "read_group"
+    //               MUST be provided as an input via a Groovy Map called "meta".
+    //               This information may not be required in some instances e.g. indexing reference genome files:
+    //               https://github.com/nf-core/modules/blob/master/modules/nf-core/bwa/index/main.nf
+    // TODO nf-core: Where applicable please provide/convert compressed files as input/output
+    //               e.g. "*.fastq.gz" and NOT "*.fastq", "*.bam" and NOT "*.sam" etc.
+    tuple val(meta), path(input_bam), path(input_bam_index)
+    tuple val(meta2), path(reference_fasta), path(reference_fasta.fai)
+    tuple val(meta3), path(temp_dir)
+
+    output:
+    // TODO nf-core: Named file extensions MUST be emitted for ALL output channels
+    tuple val(meta), path("*.vcf"), emit: vcf
```

Comment:
can you emit a gzipped vcf?
```suggestion
    tuple val(meta), path("*.vcf.gz"), emit: vcf
```

Reply:
yes, we can emit a gzip vcf file, to emit that I need to add bcftools in the `cmd` the outputs aren't gzip by the pipeline

Reply:
I think there is a way to handle this directly in the module but I am unsure how. Its just that usually we want the smallest files possible.

Reply:
I asked on slack since I did not find it directly, lets see what the other people suggest.

Reply:
I think if we pipe the .vcf file with gzip it might work.

Reply:
lets use bgzip I think that is better but do it as described on slack